js/models/messages.js

```diff
@@ -2533,9 +2560,8 @@
             }
           }
 
-          // We need to map the original message source to the primary device
           if (source !== ourNumber) {
-            message.set({ source: primarySource });
+            message.set({ primarySource });
```

Comment:
is `primarySource` the key we really want here?

Reply:
Fixed.